paas_app_charmer/utils.py

```diff
@@ -51,3 +55,52 @@ def enable_pebble_log_forwarding() -> bool:
         return True
     except ImportError:
         return False
+
+
+@functools.lru_cache
+def _config_metadata(charm_dir: pathlib.Path) -> dict:
```

Comment:
If `ops.CharmMeta` had the config (like it has actions and everything else), would that have been more convenient here? I have multiple times felt that it should have it (and once even implemented it, but that PR ended up getting discarded for unrelated reasons). Evidence of actual potential use-cases like this is interesting.

Reply:
This is required because the current `ops.CharmBase.config.get` doesn't handle `secret`-typed configurations. If the secret configs were handled by the ops library, the configuration metadata would not be necessary.